lib/routes.js

```diff
@@ -39,17 +40,23 @@ module.exports = (
   })
 
   server.route({
-    method: 'POST',
+    method: ['GET', 'POST'],
     path: config.redirectUri,
     config: {
       auth: false
     },
     handler: async (request, h) => {
       try {
-        const { payload } = request
+        const { payload, method, query } = request
+
+        if (method === 'post' && config.passRequestToCacheMethods) {
+          return h.redirect(`${config.redirectUri}?${qs.stringify(payload)}`)
+        }
+
+        const requestParams = method === 'get' ? query : payload
 
         /** Get our saved state **/
-        const { state } = payload
+        const { state } = requestParams
```

Comment:
Will the app have an unhandled crash if you do a GET without a query with 'state' in it - could happen if robot crawled site

Reply:
Query is always available as an empty object to get requests